src/RiakClient/RiakClient.cs

```diff
@@ -1653,5 +1675,28 @@ private RiakResult<TResult> UseConnection<TResult>(Func<IRiakConnection, RiakRes
                 ? op(batchConnection, () => { })
                 : endPoint.UseDelayedConnection(op, RetryCount);
         }
+
+        private bool AnyNestedRemovalsIn(IEnumerable<MapUpdate> updates)
+        {
+            foreach (var mapUpdate in updates)
+            {
+                if (mapUpdate.map_op != null && mapUpdate.map_op.removes != null && mapUpdate.map_op.removes.Count > 0)
+                {
+                    return true;
+                }
+
+                if (mapUpdate.set_op != null && mapUpdate.set_op.removes != null && mapUpdate.set_op.removes.Count > 0)
+                {
+                    return true;
+                }
+
+                if (mapUpdate.map_op != null)
+                {
+                    AnyNestedRemovalsIn(mapUpdate.map_op.updates);
```

Comment:
I think this should be:

```csharp
return AnyNestedRemovalsIn(mapUpdate.map_op.updates);
```

Otherwise the result of recursion won't be bubbled back up.